034-kafka-topic-replication-factor.md

```diff
@@ -0,0 +1,76 @@
+# Support Replication Factor Change for Kafka Topics
+
+Apache Kafka introduced the Replication Factor (RF) as the configuration that determines the number replicas kafka brokers will retain for a single data partition. A bigger number of the RF means both higher reliability for the data partition and higher infrastructure cost for topics.
+
+Apache Kafka allows users to change the RF for any topic partition through the native scripts during the runtime. However, it's not a lightweight process, because users usually need to decide which brokers to host new replicas and which replicas need to be dropped, or even conduct a cluster rebalance after the change of the RF.  
+
+The goal of this proposal is to offer users a simple approach to change the topic RF through the KafkaTopic Custom Resources and Cruise Control, in the cloud native, strimzi style. 
+
+## Current situation
+
+Currently, users have two options to change the topic RF:
+* Leverage the Native Kafka script to change the RF. Users need to generate the partition replica assignment and balance the kafka clusters manually if needed. 
+* Delete the topic and recreate the topic with the desired RF.
```

Comment:
Is there any danger of allowing users to use the native Kafka script with this newly proposed feature in place? If so, should we disable it's use somehow as part of this work?

Reply:
I think we should encourage users to use this feature to change the replication factor, instead of the Kafka native script. At the same time, it's also OK for them to use the native Kafka script, if they really want to. 

There should be no risk, unless they use both options in the same Kafka Cluster, at the same time. No matter which option the users pick, there will be one partition reassignment request that is created in the zookeeper path.

Also, is there any way to block users to use the native kafka script? Users can always ssh to kafka or zk pods and call the native scripts directly.

Reply:
In addition, if we have the cruise control running for a Kafka Cluster, it's better NOT to use the native kafka script for partition reassignment. 1) It will affect the goals of the cruise control. 2) Kafka can only execute one reassignment request at a time, so it's harder to submit any partition reassignment requests to kafka through the native Kafka script, when the cruise control is running.

So this is another reason we should encourage users not to use native kafka script for the partition reassignment tasks.